src/simulator/pysimtest/_pysimtest/suite.py

```diff
@@ -0,0 +1,207 @@
+#Created By: Logan Fillo
+#Created On: 2019-03-11
+
+"""
+This module contains the interface for creating
+simulation test suites
+"""
+
+
+import decorators
+import constants as const
+import case
+
+
+
+def forall(func):
+    """
+    Decorator for forall. Decorate inital forall
+    method with this. All methods inside decorated
+    function apply globally to a test suite.
+    Calling a method in a test method which has
+    already been called in the forall method will
+    have no effect
+
+    :param func: forall method
+    """
+    return decorators.forall(func)
+
+def test(run=True):
+    """
+    Paramaterized decorator for tests. Decorate all
+    test methods with this, including the run argument
+    specifying if you want the test to run or not
+
+    :param run: True if you want the  decorated test
+                to run, False otherwise
+    """
+    def _decorator(func):
+        return decorators.test(func, run)
+    return _decorator
+
+
+class SimTestSuite(object):
+    """
+    A class for creating simulation test suites.
+    """
+    def __init__(self):
+        """
+        Construct a SimTestSuite
+        """
+        self._in_global_state = False
+
+        self._tests = []
+        self._builder = case.SimTestBuilder()
+
+    def forall(self):
+        """
+        Abstract forall method. Use this to set test
+        parameters for all tests in the suite. Remember
+        to decorate this with forall decorater
+        """
+        pass
+
+    def use_dynamics(self, is_dynamic):
+        """
+        If is_dynamic is True, uses dynamics
+        for the test, else does not use dynamics
+
+        :param is_dynamic: if test is dynamic or not
+        """
+        # NOTE: this has not been implemented yet
```

Comment:
as in hasn't been implemented in UWsim? Or we just don't write it to the XML?

Reply:
As in the function itself hasn't been implemented. Eventually there will be a dynamics script that can be launched alongside UWsim, and the use_dynamics function will implement that. I will remove the function for now though to avoid confusion and wait until the requirements for it are met.